### PR TITLE
doc(test-features): improvements in the scorecard

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**
Some improvements in the scorecard 

**Motivation for the change:**
Try to make easier its understanding. 

**To Review:**
Check its result here: https://github.com/operator-framework/operator-sdk/blob/5ab5232a623a852e4e54ff23832926192553cd94/doc/test-framework/scorecard.md

### CODE REVIEW

**Comment by camilamacedo86** [comment]
Hi folks, 

A red hatter checked it and give the feedback that it could help understand the content better. Could we move forward with? WDYT? 

c/c @joelanford @estroz @hasbro17 @jmrodri @jmccormick2001

File: doc/test-framework/scorecard.md [hunk 1]
```diff
@@ -1,80 +1,64 @@
...
-As of `v0.8.0` scorecard also supports plugins. This allows external developers to extend the functionality of the scorecard and add additional tests.
+## Overview
 
-## Requirements
+The scorecard works by creating all resources required by CRs and the operator. 
 
-- An operator made using the `operator-sdk` or an operator that uses a config getter that supports reading from the `KUBECONFIG` environment variable (such as the `clientcmd` or `controller-runtime` config getters). This is required for the scorecard proxy to work correctly.
-- Resource manifests for installing/configuring the operator and custom resources (see the [Writing E2E Tests][writing-tests] doc for more information on the global and namespaced manifests).
-- (OLM Integration tests or Basic tests with the `olm-deployed` option set only) A CSV file for your operator.
+The scorecard will create another container in the operator deployment which is used to record calls to the API server and run a lot of the tests. These tests performe will also cover the some of the fields in the CRs objects.
```
Comment (jmccormick2001) [inline doc/test-framework/scorecard.md]: maybe should read:  "The tests performed will also examine some of the fields in the CRs."

File: doc/test-framework/scorecard.md [hunk 2]
```diff
@@ -1,80 +1,64 @@
...
 
-## Running the tests
+Since the release v0.8.0 of operator-skd, the scorecard also supports plugins which allows to extend the functionality of the scorecard and add additional tests on it.
```
Comment (jmccormick2001) [inline doc/test-framework/scorecard.md]: maybe should read:  "...the scorecard also supports plugins which allows it to extend..."

File: doc/test-framework/scorecard.md [hunk 3]
```diff
@@ -1,80 +1,64 @@
...
 
-## Running the tests
+Since the release v0.8.0 of operator-skd, the scorecard also supports plugins which allows to extend the functionality of the scorecard and add additional tests on it. 
 
-The scorecard is configured by a config file that allows configuring both internal and external plugins as well as a few global configuration options. The configuration
-options are listed below.
+## Requirements 
 
-### Flags
+Following some requirements for the operator project which would be  checked by the scorecard.
```
Comment (jmccormick2001) [inline doc/test-framework/scorecard.md]: maybe should read:  "Following are some requirements..."

File: doc/test-framework/scorecard.md [hunk 4]
```diff
@@ -1,80 +1,64 @@
...
-- `output` string - equivalent of the `--output` flag. If this option is defined by both the config file and the flag, the flag's value takes priority
-- `kubeconfig` string - equivalent of the `--kubeconfig` flag. If this option is defined by both the config file and the flag, the flag's value takes priority
-- `plugins` array - this is an array of `plugin` objects as defined below
-
-### Plugin Struct
+- Resource manifests for installing/configuring the operator and custom resources. (see the [Writing E2E Tests][writing-tests] doc for more information on the global and namespaced manifests)
+- Config getter that supports reading from the `KUBECONFIG` environment variable (such as the `clientcmd` or `controller-runtime` config getters). This is required for the scorecard proxy to work correctly.
 
-A plugin object is used to configure plugins. These are the fields of the plugin object:
+**NOTE:** If you would like to use it to check the integration of your operator project with [OLM][olm] then also the [Cluster Service Version (CSV)][olm-csv] file will be required. It is important to highlight that is a requirement just when the `olm-deployed` option is used.
```
Comment (jmccormick2001) [inline doc/test-framework/scorecard.md]: maybe rewording as:  "....This is a requirement when the `olm-deployed` option is used."

File: doc/test-framework/scorecard.md [hunk 5]
```diff
@@ -1,80 +1,64 @@
...
+- Config getter that supports reading from the `KUBECONFIG` environment variable (such as the `clientcmd` or `controller-runtime` config getters). This is required for the scorecard proxy to work correctly.
 
-A plugin object is used to configure plugins. These are the fields of the plugin object:
+**NOTE:** If you would like to use it to check the integration of your operator project with [OLM][olm] then also the [Cluster Service Version (CSV)][olm-csv] file will be required. It is important to highlight that is a requirement just when the `olm-deployed` option is used. 
 
-- `basic`, `olm`, or `external` - the configuration object for the plugin. As different types of plugins have different configuration options, they are named differently in the config. Only one of these fields can be set per plugin.
+## Running the Scorecard
 
-### Basic and OLM Internal Plugins
+1. Setup the `.osdk-scorecard config file` in your project. See [Config file](#config-file)
```
Comment (jmccormick2001) [inline doc/test-framework/scorecard.md]: maybe reword as:  "Setup the '.osdk-scorecard.yaml' configuration file in your project."

File: doc/test-framework/scorecard.md [hunk 6]
```diff
@@ -1,80 +1,64 @@
...
+2. Create the namespace defined in the RBAC files(`role_bindinding`)
+3. Then, run the operator-sdk scorecard sub-command as for example `$ operator-sdk scorecard`. See the [Command args](#command-args) to check its options.
```
Comment (jmccormick2001) [inline doc/test-framework/scorecard.md]: maybe reword as:  "Then, run the scorecard, for example '$ operator-sdk scorecard'....

File: doc/test-framework/scorecard.md [hunk 7]
```diff
@@ -110,54 +94,81 @@ The hierarchy of config methods for the global options that are also configurabl
...
+| `output` | string | equivalent of the `--output` flag. If this option is defined by both the config file and the flag, the flag's value takes priority |
+| `kubeconfig` | string | equivalent of the `--kubeconfig` flag. If this option is defined by both the config file and the flag, the flag's value takes priority |
+| `plugins` | array | this is an array of [Plugins](#plugins).|
 
-This test checks the Custom Resource(s) created in the cluster to make sure that all CRs have a status block. This
-test has a maximum score of 1.
+### Plugins
 
-#### Writing Into CRs Has An Effect
+A plugin object is used to configure plugins. The possible metadatas for the plugin object are `basic`, `olm`, or `external`.
```
Comment (jmccormick2001) [inline doc/test-framework/scorecard.md]: maybe reword:  "....The possible values for the plugin object are ...."

File: doc/test-framework/scorecard.md (hunk 1 above)
Comment (camilamacedo86) [reply doc/test-framework/scorecard.md]: Really tks. Done 👍

File: doc/test-framework/scorecard.md (hunk 2 above)
Comment (camilamacedo86) [reply doc/test-framework/scorecard.md]: I agree . really tks 👍 done.

File: doc/test-framework/scorecard.md (hunk 3 above)
Comment (camilamacedo86) [reply doc/test-framework/scorecard.md]: Good spot. Tks 👍 Done.

File: doc/test-framework/scorecard.md (hunk 4 above)
Comment (camilamacedo86) [reply doc/test-framework/scorecard.md]: Done 👍

File: doc/test-framework/scorecard.md (hunk 5 above)
Comment (camilamacedo86) [reply doc/test-framework/scorecard.md]: Perfect ! Done .

File: doc/test-framework/scorecard.md (hunk 6 above)
Comment (camilamacedo86) [reply doc/test-framework/scorecard.md]: 👍 tks Done.

File: doc/test-framework/scorecard.md (hunk 7 above)
Comment (camilamacedo86) [reply doc/test-framework/scorecard.md]: Tks 👍

**Comment by camilamacedo86** [comment]
Hi @jmccormick2001,

Really thank you for your review 🥇 . All changes requested are done.

**Comment by joelanford** [comment]
/hold

We have a bunch of upcoming changes planned for the scorecard, so I'm not sure it makes sense to spend much time updating and reviewing the documentation for the current implementation unless there's something specifically incorrect or misleading about it.

I'll defer to @jmccormick2001 and @dmesser if they would find it helpful to have these updates made now.

**Comment by camilamacedo86** [comment]
Hi @joelanford, 

The few changes performed here are helpful for the users who try to understand this feature currently. I redhater use this PR yesterday and tell me that he could understand much easier how it works with. 

Note that we just supplement a few info, and improve the sequence of the info and add the tables which makes easier check the content. It should is not really L in POV. (The L here is just because of putting the same content in tables and change a little the order of the info)

In order to advocate for we merge this PR the scorecard will change but also it will take a while and I think we will be updating this doc incrementally so at the end the changes performed n POV will still helpful.

**Comment by camilamacedo86** [comment]
Hi @dmesser,

@jmccormick2001 already approved before this PR. Have you any objection about these improvements or could we move forward here? 

Thank you for your attention and time.

**Comment by camilamacedo86** [comment]
hi @joelanford, 

I am removing the "hold" label since @dmesser and @jmccormick2001 are ok with the changes here. Also, after we merge it I will try to help on  https://github.com/operator-framework/operator-sdk/pull/1816. I hope that these changes will help us to organize better the content of this one too.
